rust-runtime/aws-smithy-runtime-api/src/type_erasure.rs

```diff
@@ -62,75 +62,84 @@ where
     }
 }
 
-impl<T: 'static> Deref for TypedBox<T> {
+impl<T> fmt::Debug for TypedBox<T>
+where
+    T: fmt::Debug + Send + Sync + 'static,
```

Comment:
I don't think you need this restriction—it has to be `T` to go into the type erased box, you can delegate the call directly to the debug impl of type erased box